pyzebra/load_1D.py

```diff
@@ -0,0 +1,172 @@
+import re
+from collections import defaultdict
+from decimal import Decimal
+
+import numpy as np
+
+META_VARS_STR = (
+    "instrument",
+    "title",
+    "sample",
+    "user",
+    "ProposalID",
+    "original_filename",
+    "date",
+    "zebra_mode",
+    "proposal",
+    "proposal_user",
+    "proposal_title",
+    "proposal_email",
+)
+META_VARS_FLOAT = (
+    "mf",
+    "2-theta",
+    "chi",
+    "phi",
+    "nu",
+    "temp",
+    "wavelenght",
+    "a",
+    "b",
+    "c",
+    "alpha",
+    "beta",
+    "gamma",
+    "cex1",
+    "cex2",
+    "mexz",
+    "moml",
+    "mcvl",
+    "momu",
+    "mcvu",
+    "detectorDistance",
+    "snv",
+    "snh",
+    "snvm",
+    "snhm",
+    "s1vt",
+    "s1vb",
+    "s1hr",
+    "s1hl",
+    "s2vt",
+    "s2vb",
+    "s2hr",
+    "s2hl",
+)
+META_UB_MATRIX = ("ub1j", "ub2j", "ub3j")
+
+
+def load_1D(filepath):
+    """
+    Loads *.ccl or *.dat file (Distinguishes them based on last 3 chars in string of filepath
+    to add more variables to read, extend the elif list
+    the file must include '#data' and number of points in right place to work properly
+
+    :arg filepath
+    :returns det_variables
+    - dictionary of all detector/scan variables and dictinionary for every measurement.
+    Names of these dictionaries are M + measurement number. They include HKL indeces, angles,
+    monitors, stepsize and array of counts
+    """
+    det_variables = {"file_type": str(filepath)[-3:], "meta": {}}
+    with open(filepath, "r") as infile:
+        # read metadata
+        for line in infile:
+            det_variables["Measurements"] = {}
+            if "=" in line:
+                variable, value = line.split("=")
+                variable = variable.strip()
+                if variable in META_VARS_FLOAT:
+                    det_variables["meta"][variable] = float(value)
+                elif variable in META_VARS_STR:
+                    det_variables["meta"][variable] = str(value)[:-1].strip()
+                elif variable in META_UB_MATRIX:
+                    det_variables["meta"][variable] = re.findall(r"[-+]?\d*\.\d+|\d+", str(value))
+
+            if "#data" in line:
+                # this is the end of metadata and the start of data section
+                break
+
+        # read data
+        if det_variables["file_type"] == "ccl":
+            decimal = list()
+            data = infile.readlines()

```

Comment:
@JakHolzer , if you want, you can try to simplify reading of ccl files in a similar way to https://github.com/paulscherrerinstitute/pyzebra/pull/11/commits/fe1d8044a3f2f6e24927a4f8baf4aa0894a56140

Reply:
I will try 👍

Reply:
Btw, that commit also renames "omega" into "om", and "counts" into "Counts", that is consistent with the column names in dat files. I hope it doesn't break you other code, otherwise let me know what names you prefer, it will be easy to fix, e.g. with
```
# row_names = next(infile).split()
row_names = ["NP", "omega", "counts", ...]
```